src/components/MenuProductModal/MenuProductModal.tsx

```diff
@@ -0,0 +1,125 @@
+import { ReactElement, useCallback, useState } from 'react';
+import { ProductDto } from 'dtos/ProductDto';
+import { Button } from 'components/Button/Button';
+import { useLocalOrder } from 'context/OrderContext';
+import PlaceHolderImage from 'assets/burger_figma.png';
+
+interface Props {
+  product: ProductDto;
+  showModal: boolean;
+  disableShow: (isShow: boolean) => void;
+}
+
+export const MenuProductModal = ({ product, showModal, disableShow }: Props): ReactElement => {
+  const { addToLocalOrder, removeProduct, localOrder } = useLocalOrder();
+
+  const checkQuantity = (product: ProductDto): number => {
+    const existingItem = localOrder.lineItems.find((item) => item.product.id === product.id);
+    return existingItem ? existingItem.quantity : 1;
+  };
+
+  const [counter, setCounter] = useState(checkQuantity(product));
+  const increase = (): void => {
+    setCounter((counter) => counter + 1);
+  };
+
+  const decrease = (): void => {
+    if (counter) {
+      setCounter((counter) => counter - 1);
+    }
+  };
+
+  const addToOrder = useCallback(() => {
+    addToLocalOrder(product, counter);
+  }, [addToLocalOrder, product, counter]);
+
+  const allergens = ['Milk', 'Peanuts', 'Celery', 'Soy', 'Gluten'];
+
+  return (
+    <>
+      {showModal ? (
+        <>
+          <div
+            className="fixed inset-0 z-50 flex items-center justify-center overflow-x-auto overflow-y-auto
+       bg-black bg-opacity-25 outline-none focus:outline-none"
+            onClick={() => disableShow(false)}
+          >
+            <div
+              className="relative flex h-4/5 w-5/6 flex-col items-center justify-between rounded-2xl bg-white p-6 shadow-[0_4px_4px_rgba(0,0,0,0.25)] lg:h-fit lg:w-3/5 xl:w-[35%]"
+              onClick={(e) => e.stopPropagation()}
+            >
+              <div className="flex flex-row gap-6">
+                <img alt="burger" className="h-[172.41px] w-[264px] grow rounded-[12px] border-0 p-2" src={PlaceHolderImage} />
+                <div className="flex flex-col items-start gap-2 p-4">
+                  <h5 className="font-poppins text-xl font-[600] text-[#333F4C]">{product.name}</h5>
```

Comment:
use px for text - i.e text-[16px], easier to read for people coming onto the project